packages/library/src/commonMain/kotlin/io/realm/RealmList.kt

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.interop.NativePointer
+
+/**
+ * TODO
+ */
+class RealmList<E> : MutableList<E>, AbstractMutableCollection<E> {
```

Comment:
Just for completion: I somehow feel that this is a bit verbose, but of discussion I guess we cannot use delegation from inheritance (as it only works for interfaces and you would eliminate the option of extending `AbstractMutableCollection`) and we want to do the differentiation between managed and unmanaged internally, as having two public types pollutes the public space.